include/quicr/detail/cache_buffer.h

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024 Cisco Systems
+// SPDX-License-Identifier: BSD-2-Clause
+
+#pragma once
+
+#include "span.h"
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+class CacheBuffer
+{
+    template<class It, class SpanIt>
+    class IteratorImpl
+    {
+        using it_traits = std::iterator_traits<const std::uint8_t*>;
+        using it_impl = IteratorImpl;
+
+      public:
+        using value_type = typename it_traits::value_type;
```

Comment:
Is there a clean way to indicate our styling is ignored here?

Reply:
I'm going to remove these, I just like to code STL style containers like STL.